docs/modules/dimmer.md

```diff
@@ -0,0 +1,135 @@
+# dimmer Module
+| Since      | Origin / Contributor                            | Maintainer                                      | Source                                        |
+| :--------- | :---------------------------------------------- | :---------------------------------------------- | :-------------------------------------------- |
+| 2020-01-29 | [Javier Peletier](https://github.com/jpeletier) | [Javier Peletier](https://github.com/jpeletier) | [dimmer.c](../../components/modules/dimmer.c) |
+
+This module implements phase-dimming for TRIAC-based mains dimmers. 
+
+## How phase-dimming works
+Find here a [brief description on how TRIAC-based dimming works](https://www.lamps-on-line.com/leading-trailing-edge-led-dimmers).
+
+In essence, phase dimming implies generating a square-wave PWM signal, with the special characteristic that it must be perfectly synchronized with mains, modulating it, and thus delivering more or less effective power to the load depending on the PWM signal duty cycle.
+
+In order to do this, a dimmer module or circuit must be used to a) detect when the mains signal crosses 0V, to use it as a synchronization point and b) modulate mains. The module must isolate mains from the microcontroller.
+
+Some reference hardware: 
+* [Schematics for a homemade module](https://hackaday.io/project/165927-a-digital-ac-dimmer-using-arduino/details)
+* [Example commercial hardware](https://robotdyn.com/ac-light-dimmer-module-1-channel-3-3v-5v-logic-ac-50-60hz-220v-110v.html)
+
+These modules come with a TRIAC whose gate is intended to be driven by a GPIO pin, isolated from mains by an optocoupler. These modules also come with a zero-crossing detector, also isolated, that raises a pin voltage when the AC mains sine wave signal crosses 0V.
+
+## Architecture of this nodemcu module
+
+The above scheme is implemented in this module by dedicating ESP32's CPU1 entirely to this purpose... Phase dimming requires very accurate timing. Configuring timer interrupts in the "busy" CPU0 does not work properly, with FreeRTOS scheduler, WiFi and so on demanding their share of the CPU at random intervals, which would make dimmed lamps flicker.
+
+Once the dimmer module is started, by means of `dimmer.setup()`, a busy loop is launched on CPU1 that monitors zero-crossing signals from the dimmer and turns on/off the TRIAC at the appropriate time, with nanosecond precision.
+
+## Required SDK `menuconfig` changes
+To use this module, change the following in menuconfig (`make menuconfig`):
+
+* Enable *FreeRTOS in both cores* by unselecting *"Component Config/FreeRTOS/Run FreeRTOS only on first core"*. This will allow the module to use CPU1.
+* Unselect  *"Component Config/ESP32-specific/Also watch CPU1 tick interrupt"*
+* Unselect  *"Component Config/ESP32-specific/Watch CPU1 idle task"*
```

Comment:
```suggestion
* Enable FreeRTOS in both cores by **deselecting** `Component Config` > `FreeRTOS` > `Run FreeRTOS only on first core`. This will allow the module to also use CPU1 (besides CPU0).
* **Deselect**  `Component Config` > `ESP32-specific` > `Also watch CPU1 tick interrupt`
* **Deselect**  `Component Config` > `ESP32-specific` > `Watch CPU1 idle task`
```